lib/graphql/generators/install_generator.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+require 'rails/generators/base'
+
+module GraphQL
+  module Generators
+    # Add GraphQL to a Rails app with
+    # `rails g graphql:install`.
+    #
+    # Setup a folder structure for GraphQL:
+    #
+    # ```
+    # - app/
+    #   - graphql/
+    #     - resolvers/
+    #     - types/
+    #       - query_type.rb
+    #     - loaders/
+    #     - mutations/
+    #     - {app_name}_schema.rb
+    # ```
+    #
+    # (Add `.gitkeep`s by default, support `--skip-keeps`)
+    #
+    # Add a controller for serving GraphQL queries:
+    #
+    # ```
+    # app/controllers/graphql_queries_controller.rb
+    # ```
+    #
+    # Add a route for that controller:
+    #
+    # ```ruby
+    # # config/routes.rb
+    # post "/graphql", to: "graphql_queries#create"
+    # ```
+    #
+    # Accept a `--relay` option which adds
+    # The root `node(id: ID!)` field.
+    #
+    # Accept a `--graphiql` option which adds
+    # `graphiql-rails` to the Gemfile and mounts the
+    # engine in `routes.rb` (Should this be the default?)
+    class InstallGenerator < Rails::Generators::Base
+      desc "Install GraphQL folder structure and boilerplate code"
+      source_root File.expand_path('../templates', __FILE__)
+
+      class_option :schema,
+        type: :string,
+        default: nil,
+        desc: "Name for the schema constant (default: {app_name}Schema)"
+
+      class_option :skip_keeps,
+        type: :boolean,
+        default: false,
+        desc: "Skip .keep files for source control"
+
+      class_option :skip_graphiql,
+        type: :boolean,
+        default: false,
+        desc: "Skip graphiql-rails installation"
+
+      class_option :relay,
+        type: :boolean,
+        default: false,
+        desc: "Include GraphQL::Relay installation"
+
+      class_option :batch,
+        type: :boolean,
+        default: false,
+        desc: "Include GraphQL::Batch installation"
+
+
+      GRAPHIQL_ROUTE = <<-RUBY
+if Rails.env.development?
+    mount GraphiQL::Rails::Engine, at: "/graphiql", graphql_path: "/graphql"
+  end
+RUBY
+
+      def create_folder_structure
+        create_dir("app/graphql/mutations")
+        create_dir("app/graphql/resolvers")
+        create_dir("app/graphql/types")
+        template("query_type.erb", "app/graphql/types/query_type.rb")
+        template("schema.erb", "app/graphql/#{schema_name.underscore}.rb")
+        template("graphqls_controller.erb", "app/controllers/graphqls_controller.rb")
```

Comment:
Could be, then we'd have to switch:

```diff 
- resource :graphql, only: :create 
+ post "/graphql", to: "graphql#create"
```

I guess it's not really a resource anyways, old habits die hard :P